statesync/stateprovider.go

```diff
@@ -96,6 +97,18 @@ func NewLightClientStateProvider(
 	}, nil
 }
 
+func uniqServers(servers []string) []string {
+	serversSet := make(map[string]struct{})
```

Comment:
nit, it could be better to specify the initial capacity: `make(map[string]struct{}, len(servers))`